trees/redblacktree/redblacktree.go

```diff
@@ -126,6 +126,64 @@ func (tree *Tree) Get(key interface{}) (value interface{}, found bool) {
 	return nil, false
 }
 
+// Find ceiling node of the input key, return its key and value or nil if no ceiling is found.
+// Third return parameter is true if ceiling was found, otherwise false.
+// Ceiling node is defined as the smallest node that is larger than or equal to the given node.
+// A ceiling node may not be found, either because the tree is empty, or because
+// all nodes in the tree is smaller than the given node.
+// Key should adhere to the comparator's type assertion, otherwise method panics.
+func (tree *Tree) Ceiling(key interface{}) (ceilingKey interface{}, value interface{}, found bool) {
```

Comment:
This can be simplified to two return parameters: node and found. Node is exposed now and already contains the key and the value.